### PR TITLE
Allow for single named custom events in reporting client

### PR DESCRIPTION
**Pull request by ashzagorski**

### CODE REVIEW

**Comment by ashzagorski** [comment]
[Allow for single named custom events in reporting client](https://trello.com/c/7yFPTFvO/3556-allow-for-single-named-custom-events-in-reporting-client)

File: lib/reporting_client/event.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'active_support'
+require 'rubygems'
+
+module ReportingClient
+  class Event
+    cattr_accessor :events, default: []
```
Comment (jorin) [inline lib/reporting_client/event.rb:8]: we don't need this since there's only the one event for this interface, I think ...

**Review by jorin — COMMENTED** [review]
It feels like there's room to DRY this up with maybe an abstract super class, since `Event` and `Events` have the same code for `instrument` and `send_event`. Actually `Events` just extends `Event`, so maybe `Event` is the super class for `Events`? Not sure, but I feel like we should sort that out if not in this PR, soon.

But we could address that with a follower, I think the only thing we should do here is make sure to include, I guess just in the documentation for the single event name stuff of the `README`, how to do the `ActiveSupport::Notifications.subscribe` for that event. Otherwise somebody may grab this, just set up the name, and not get any instrumentation b/c they didn't subscribe their event.

**Review by jorin — APPROVED** [review]
+1 _but also_ take a look at my comment there. This works and DRYs it up, but I think the relationship between the classes is inverted: with `Event` inheriting from `Events`, it gets a bunch of stuff it doesn't need (which ends up being fine b/c we won't use it), but I think what we really want here is `Events` inheriting from `Event`, b/c `Event` can be simpler, and `Events` adds all the multiple event type handling as an augmentation to that behavior.

**Comment by ashzagorski** [comment]
@jorin No worries, thanks for the feedback! Can you just merge this in? It is currently not being consumed so I don't think any QA is necessary for the time being. There is still some discussion around reporting in glue so not ready for a PR there yet but when we do pull it in there we can QA at that time. .

**Comment by jorin** [comment]
yep will do ...

**Pull request merged**